samcli/lib/telemetry/project_metadata.py

```diff
@@ -88,24 +94,46 @@ def get_initial_commit_hash() -> Optional[str]:
     except subprocess.CalledProcessError:
         return None  # Not a git repo
 
-    return _encrypt_value(metadata)
+    return _hash_value(metadata)
 
 
-def _parse_remote_origin_url(url: str) -> List[str]:
+def _parse_remote_origin_url(url: str) -> Optional[str]:
     """
-    Parse a `git remote origin url` into its hostname, owner, and project name.
+    Parse a `git remote origin url` into a formatted "hostname/project" string
 
     Returns
     -------
-    List[str]
-        A list of 3 strings, with indeces corresponding to 0:hostname, 1:owner, 2:project_name
+    str
+        formatted project origin url
     """
-    pattern = re.compile(r"(?:https?://|git@)(?P<hostname>\S*)(?:/|:)(?P<owner>\S*)/(?P<project_name>\S*)\.git")
-    return [str(item) for item in pattern.findall(url)[0]]
+    found = None
+    # Examples: http://example.com:8080/git/repo.git, https://github.com/aws-actions/setup-sam/
+    http_pattern = re.compile(r"(\w+://)(.+@)*(?P<url>([\w\.]+)(:[\d]+)?/*(.*))")
+    http_match = http_pattern.match(url)
+    if http_match:
+        found = http_match.group("url")
+    # Examples: git@github.com:aws/serverless-application-model.git, git@example.com:my/repo
+    git_ssh_pattern = re.compile(r"(git@)(?P<url>([\w\.]+):(.*))")
+    git_ssh_match = git_ssh_pattern.match(url)
+    if not found and git_ssh_match:
+        found = git_ssh_match.group("url")
+
+    # NOTE(hawflau): there are other patterns that we don't handle now, as they are not the most common use case
+    # cleaning and formatting
+    if not found:
+        return None
+    if found.endswith("/"):
+        found = found[:-1]
+    if found.endswith(".git"):
+        found = found[:-4]
+    found = re.sub(r":\d{1,5}", "", found)  # remove port number if any
+    found = found.replace(":", "/")  # e.g. github.com:aws/aws-sam-cli -> github.com/aws/aws-sam-cli
+
+    return found
```

Comment:
This can be replaced with:

```python
import re
from urllib.parse import urlparse

def parse(url):
    u = urlparse(url)
    s = (u.hostname or "") + u.path
    s = re.sub("/$", "", s)
    s = re.sub(".git$", "", s)
    s = re.sub("^git@", "", s)
    s = s.replace(":", "/")
    return s
```

And it'll pass all the test cases in `test_retrieve_git_origin`.

Reply:
Thanks for the suggestion! Will update. 👍